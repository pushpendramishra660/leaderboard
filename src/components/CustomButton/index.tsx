import React from 'react';
import styled from 'styled-components/native';
import theme from '../../theme';

const StyledButtonContainer = styled.TouchableOpacity`
  background-color: ${theme.colors.blue};
  padding: 10px 20px;
  border-radius: 5px;
`;

const StyledButtonText = styled.Text`
  color: ${theme.colors.white};
  font-size: 16px;
  text-align: center;
`;

interface CustomButtonProps {
  label: string;
  onPress: () => void;
}

const CustomButton: React.FC<CustomButtonProps> = ({label, onPress}) => {
  return (
    <StyledButtonContainer onPress={onPress}>
      <StyledButtonText>{label}</StyledButtonText>
    </StyledButtonContainer>
  );
};

export default CustomButton;
