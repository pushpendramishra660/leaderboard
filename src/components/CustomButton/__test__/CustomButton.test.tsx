import React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';
import CustomButton from '../index';

describe('<CustomButton />', () => {
  it('renders button correctly', () => {
    const {getByText} = render(
      <CustomButton label="Test Button" onPress={() => {}} />,
    );
    const buttonElement = getByText('Test Button');
    expect(buttonElement).toBeTruthy();
  });
  it('calls the provided function when button is pressed', () => {
    const onPressMock = jest.fn();
    const {getByText} = render(
      <CustomButton label="Test Button" onPress={onPressMock} />,
    );
    const buttonElement = getByText('Test Button');
    fireEvent.press(buttonElement);
    expect(onPressMock).toHaveBeenCalled();
  });
  it('displays correct button label', () => {
    const {getByText} = render(
      <CustomButton label="Submit" onPress={() => {}} />,
    );
    const buttonElement = getByText('Submit');
    expect(buttonElement).toBeTruthy();
  });
  it('CustomButton component matches snapshot', () => {
    const onPressMock = jest.fn();
    const component = renderer.create(
      <CustomButton label="Test Button" onPress={onPressMock} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
