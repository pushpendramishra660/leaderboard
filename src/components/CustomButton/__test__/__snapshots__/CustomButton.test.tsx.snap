// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> CustomButton component matches snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#007bff",
      "borderBottomLeftRadius": 5,
      "borderBottomRightRadius": 5,
      "borderTopLeftRadius": 5,
      "borderTopRightRadius": 5,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    Test Button
  </Text>
</View>
`;
